sdk/storage/azure-storage-files-datalake/src/datalake_file_client.cpp

```diff
@@ -157,13 +157,11 @@ namespace Azure { namespace Storage { namespace Files { namespace DataLake {
     protocolLayerOptions.RetainUncommittedData = options.RetainUncommittedData;
     protocolLayerOptions.Close = options.Close;
     protocolLayerOptions.ContentLength = 0;
-    if (options.ContentHash.HasValue())
+    if (options.ContentHash.HasValue()
+        && options.ContentHash.Value().Algorithm == HashAlgorithm::Md5)
     {
-      AZURE_ASSERT_MSG(
-          options.ContentHash.Value().Algorithm == HashAlgorithm::Md5,
-          "This operation only supports MD5 content hash.");
+      protocolLayerOptions.ContentMd5 = options.ContentHash;
     }
-    protocolLayerOptions.ContentMd5 = options.ContentHash;
```

Comment:
Why is this not going to the server and letting the server reject it?

cc: @JeffreyRichter, @tg-msft

Reply:
Yes, I think the header should just be set regardless and the service deals with it.

Reply:
> Why is this not going to the server and letting the server reject it?

1. Server won't reject it, server ignores this header.

2. Protocol layer code is auto-generated based on swagger (or something works like swagger). There's no way to add some headers that are not defined in swagger.

Reply:
> Yes, I think the header should just be set regardless and the service deals with it.

And what the HTTP header name would be?
What if service team decides to use a different HTTP header name when they support this feature?